### PR TITLE
Added 'page' state in the redux store along with configuring the navb…

### PR DESCRIPTION
**Pull request by nnmkhang**
Overview of Changes in this PR:
- Added state "page" to the redux store
- Added Dynamic page name rendering ( "Dashboard" can now be changed by modifying the page sate in the redux store )
- Fixed minor formatting issues 
- Added small CSS changes that include:
  - Navbar from grey -> blue
  - Margin difference from dashboard and navbar
  - Font size
  - Adjusted Searchbar size  
  - Logo Position

Margin Before: 
image

Margin After: 
image

Font before: 
image

Font after: 
image

Search Bar before:
image

Search Bar after:
image

Logo Position before: 
image

Logo Position after:
image

### CODE REVIEW

File: private-templates-service/client/src/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+  currentPage: string;
+}
+
+class NavBar extends React.Component<NavBarProps, {}> {
+  render(){
+    return (
+      <Banner>
+        <MobileBanner>
+          <StyledLogo src = {Logo}/>
+          <Styledh1> {this.props.currentPage} </Styledh1>
```
Comment (kodyang) [inline private-templates-service/client/src/components/NavBar/NavBar.tsx]: can you get rid of the whitespace before and after the `{...}`

File: private-templates-service/client/src/store/rootReducer.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import { authReducer } from './auth/reducers';
 import { searchReducer } from './search/reducer';
+import { pageReducer } from './page/reducer';
 import { combineReducers } from 'redux';
 
 export const rootReducer = combineReducers({
   auth: authReducer,
   search: searchReducer,
+  page: pageReducer
```
Comment (kodyang) [inline private-templates-service/client/src/store/rootReducer.ts]: add a comma at the end

File: private-templates-service/client/src/components/Dashboard/styled.tsx [hunk 3]
```diff
@@ -6,5 +6,5 @@ export const Title = styled.h1`
 `;
 
 export const DashboardContainer = styled.div`
-	margin-left: 30px;
+	margin: 0px 40px;
```
Comment (kodyang) [inline private-templates-service/client/src/components/Dashboard/styled.tsx]: `margin: 0 40px;`

File: private-templates-service/client/src/components/NavBar/NavBar.tsx [hunk 4]
```diff
@@ -1,18 +1,32 @@
...
+interface NavBarProps { 
+  currentPage: string;
+}
+
+class NavBar extends React.Component<NavBarProps, {}> {
+  render(){
+    return (
+      <Banner>
+        <MobileBanner>
+          <StyledLogo src = {Logo}/>
```
Comment (kodyang) [inline private-templates-service/client/src/components/NavBar/NavBar.tsx]: get rid of whitespace around the `=`

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx [hunk 5]
```diff
@@ -15,17 +15,28 @@ const mapStateToProps = (state: RootState) => {
...
+      dispatch(setPage(currentPage));
+    }
+  }
+}
+
+class Dashboard extends React.Component<DashboardProps, {}> {
+
+  constructor(props: DashboardProps){
+    super(props);
+    props.setPage("Dashboard");
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: Do you want every time a Dashboard component is created to change the page to dashboard?

File: private-templates-service/client/src/store/page/actions.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { PageAction, NAVIGATION } from "./types";
+
+export function setPage(currentPage: string): PageAction{
+  return{
+    type: NAVIGATION,
+    text: "The user has changed the page",
```
Comment (Jackbk) [inline private-templates-service/client/src/store/page/actions.ts]: Unless I'm missing something I dont think you need descriptions in your actions

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx (hunk 5 above)
Comment (Jackbk) [reply private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: Oh I see, this is just the title.  Not the rendering children.

File: private-templates-service/client/src/store/page/types.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+// Action Types
+export const NAVIGATION = "NAVIGATION";
+
+// Actions 
+export interface PageAction{
+  type: typeof NAVIGATION;
+  text: string;
+  currentPage: string;
```
Comment (Jackbk) [inline private-templates-service/client/src/store/page/types.ts]: currentPageTitle

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx (hunk 5 above)
Comment (nnmkhang) [reply private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: Yes, we are only rendering the title page name, I will add some comments to make this more clear

File: private-templates-service/client/src/store/page/actions.ts (hunk 6 above)
Comment (nnmkhang) [reply private-templates-service/client/src/store/page/actions.ts]: We're leaving this in to make it easier to debug with the redux plugin on chrome

File: private-templates-service/client/src/store/page/types.ts (hunk 7 above)
Comment (nnmkhang) [reply private-templates-service/client/src/store/page/types.ts]: I changed all instances of currentPage -> currentPageTitle to make it more clear that the title page is rendering

**Pull request merged**